driver-mapping/src/main/java/com/datastax/driver/mapping/MappedUDTCodec.java

```diff
@@ -87,13 +87,35 @@ protected T deserializeAndSetField(ByteBuffer input, T target, String fieldName,
 
     @Override
     protected String formatField(T source, String fieldName) {
-        // This codec implementation is internal-use only, and its format method is never used
-        throw new UnsupportedOperationException();
+        if (!fieldName.startsWith("\""))
+            fieldName = Metadata.quote(fieldName);
+
+        ColumnMapper<T> columnMapper = columnMappers.get(fieldName);
+
+        if (columnMapper == null)
+            return null;
+
+        Object value = columnMapper.getValue(source);
+
+        TypeCodec<Object> codec = columnMapper.getCustomCodec();
+        if (codec == null)
+            codec = codecRegistry.codecFor(cqlUserType.getFieldType(columnMapper.getColumnName()), columnMapper.getJavaType());
+
+        return codec.format(value);
     }
 
     @Override
     protected T parseAndSetField(String input, T target, String fieldName) {
-        // This codec implementation is internal-use only, and its parse method is never used
-        throw new UnsupportedOperationException();
+        if (!fieldName.startsWith("\""))
```

Comment:
In this case it isn't escaped before passing it in, although maybe that should be done in `AbstractUDTCodec#parse`